CMakeLists.txt

```diff
@@ -567,23 +576,33 @@ else()
 endif()
 
 target_compile_features(torrent-rasterbar PUBLIC cxx_std_11)
-
-target_compile_definitions(torrent-rasterbar PUBLIC $<$<CONFIG:Debug>:TORRENT_DEBUG>)
-
-target_compile_definitions(torrent-rasterbar PRIVATE TORRENT_BUILDING_LIBRARY)
+target_compile_definitions(torrent-rasterbar
+	PUBLIC
+		$<$<CONFIG:Debug>:TORRENT_DEBUG>
+		$<$<CONFIG:Debug>:TORRENT_USE_ASSERTS>
+	PRIVATE
+		TORRENT_BUILDING_LIBRARY
+)
+target_link_libraries(torrent-rasterbar
+	PUBLIC
+		Threads::Threads
+)
 
 if (libiconv)
 	find_package(Iconv REQUIRED)
+	list(APPEND _package_dependencies "Iconv")
 	target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_ICONV)
-	target_include_directories(torrent-rasterbar SYSTEM PUBLIC ${ICONV_INCLUDE_DIR})
+	target_include_directories(torrent-rasterbar PUBLIC ${ICONV_INCLUDE_DIR})
+	target_link_libraries(torrent-rasterbar PRIVATE ${ICONV_LIBRARIES})
 endif (libiconv)
 
 if (encryption)
-	if(NOT DEFINED OPENSSL_INCLUDE_DIR OR NOT DEFINED OPENSSL_LIBRARIES)
-		FIND_PACKAGE(OpenSSL REQUIRED)
+	if(NOT TARGET OpenSSL::SSL)
+		find_package(OpenSSL REQUIRED)
 	endif()
+	list(APPEND _package_dependencies "OpenSSL")
+	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
```

Comment:
is `OpenSSL::SSL` automatically pulling in the dependency on `OpenSSL::Crypto`?

Reply:
It does in CMake 3.11, and I believe it does the same in other versions too.

Reply:
Yep, checked with their Git.